### PR TITLE
Ensure files are pdfs

### PR DESCRIPTION
**Pull request by olaughter**
This adds test coverage for api_client and fixes two issues:
1. Uploading invalid pdfs to our cdn leads to errors, so we need to make sure they are valid. Rather then reinventing the wheel ourselves we can make use of pypdf's reader. Invalid files will fail with `PyPdfError` which we catch
2. The content type was written before validation checks, this meant invalid pdfs could still be flagged as pdfs

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -24,6 +24,8 @@ requests = "^2.28.1"
 tenacity = "^8.1.0"
 json-logging = "^1.3.0"
 cpr-data-access = {git = "https://github.com/climatepolicyradar/data-access.git", tag = "0.4.6"}
+pypdf = "^4.2.0"
+requests-mock = "^1.12.1"
```
Comment (THOR300) [inline pyproject.toml:28]: Should this be in the dev deps?

File: src/navigator_data_ingest/base/api_client.py [hunk 2]
```diff
@@ -55,6 +59,7 @@ def upload_document(
     try:
         download_response = _download_from_source(session, source_url)
         content_type = determine_content_type(download_response, source_url)
+        file_content = download_response.content
```
Comment (THOR300) [inline src/navigator_data_ingest/base/api_client.py:62]: Can't this move to line 73 for clarity.

File: src/navigator_data_ingest/tests/conftest.py [hunk 3]
```diff
@@ -272,3 +272,39 @@ def test_updates(s3_document_id):
         )
         for update in updates[s3_document_id]
     ]
+
+
+@pytest.fixture
+def mock_cdn_config():
+    """CDN bucket config to be used for non-aws based tests"""
+    return {
+            "bucket": "offline-unit-tests-cdn",
```
Comment (THOR300) [inline src/navigator_data_ingest/tests/conftest.py:281]: Indentation here looks off!

File: src/navigator_data_ingest/tests/conftest.py [hunk 4]
```diff
@@ -272,3 +272,39 @@ def test_updates(s3_document_id):
...
+            "region": "eu-west-1",
+    }
+
+
+@pytest.fixture
+def test_s3_client__cdn(mock_cdn_config):
+    """Empty csn and pipeline buckets to be used for non-aws based testing"""
```
Comment (THOR300) [inline src/navigator_data_ingest/tests/conftest.py:288]: Csn should be cdn

**Review by THOR300 — APPROVED** [review]
Some non blocking comments.

File: pyproject.toml (hunk 1 above)
Comment (olaughter) [reply pyproject.toml:28]: Yep

File: src/navigator_data_ingest/base/api_client.py (hunk 2 above)
Comment (olaughter) [reply src/navigator_data_ingest/base/api_client.py:62]: Sure

File: src/navigator_data_ingest/tests/conftest.py (hunk 3 above)
Comment (olaughter) [reply src/navigator_data_ingest/tests/conftest.py:281]: Good spot

File: src/navigator_data_ingest/tests/conftest.py (hunk 4 above)
Comment (olaughter) [reply src/navigator_data_ingest/tests/conftest.py:288]: fixed

**Pull request merged**